src/docdb/tree/DocDBStoredProceduresTreeItem.ts

```diff
@@ -69,14 +71,29 @@ export class DocDBStoredProceduresTreeItem extends DocDBTreeItemBase<StoredProce
         return this.parent.getContainerClient(client);
     }
 
-    private validateName(name: string): string | null | undefined {
+    private async validateStoredProcedureName(name: string, getChildrenTask: Promise<AzExtTreeItem[]>): Promise<string | null | undefined> {
+        const currStoredProcedureList = await getChildrenTask;
+        const currStoredProcedureNames: string[] = [];
+        for (const sp of currStoredProcedureList) {
+            if (sp instanceof DocDBStoredProcedureTreeItem) {
+                currStoredProcedureNames.push(sp.id);
+            }
+        }
         if (name) {
-            if (name.indexOf("/") !== -1 || name.indexOf("\\") !== -1 || name.indexOf("?") !== -1 || name.indexOf("#") !== -1) {
-                return "Id contains illegal chars: /,\\,?,#";
+            if (name.length < 1 || name.length > 255) {
+                return "Name has to be between 1 and 255 chars long";
+            }
+            if (/[/\\?#&]/.test(name)) {
+                return localize("illegalChars", "Name contains illegal chars: /,\\,?,#,&");
```

Comment:
From https://github.com/microsoft/vscode-cosmosdb/issues/1719:

> Succeed to create a new stored procedure with name as "&&&" without any error in portal.

It sounds like we should allow `&` if the portal does

Reply:
The portal throws an error while creating a sp containing "&" in the name. Even if the extension allows creating sp with "&" in the name, there is an error while deleting it in the portal. 
![image](https://user-images.githubusercontent.com/17537419/107436537-83200900-6ae2-11eb-8ad9-22ecd1ac6451.png)